### PR TITLE
Feat: local virtual cluster + namespace in topology

### PR DESCRIPTION
**Pull request by Somefive**
Signed-off-by: Somefive <yd219913@alibaba-inc.com>

### Description of your changes

1. Allow local cluster as VirtualCluster. Now user can use local cluster in topology policy like previous env-binding policy.
2. Allow specifying namespace in topology policy. Now user can use a different namespace other than the original application namespace like previous env-binding policy. Notice that this will only work when bootstrap parameter --allow-cross-namespace-resource is set.

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### Special notes for your reviewer

Picked from #3409.

### CODE REVIEW

**Review by wonderflow — APPROVED** [review]

File: apis/core.oam.dev/v1alpha1/policy_types.go [hunk 1]
```diff
@@ -27,6 +27,8 @@ const (
 type TopologyPolicySpec struct {
 	Clusters        []string          `json:"clusters,omitempty"`
 	ClusterSelector map[string]string `json:"clusterSelector,omitempty"`
+	// Namespace specify the namespace for override, used along which clusters/clusterLabelSelector/clusterSelector
```
Comment (yue9944882) [inline apis/core.oam.dev/v1alpha1/policy_types.go:30]:
```
// Namespace is the target namespace to deploy in the selected clusters.
// +optional
```

nit: clarifying the optionality and revise the godoc

File: pkg/multicluster/virtual_cluster.go [hunk 2]
```diff
@@ -49,6 +45,18 @@ type VirtualCluster struct {
 	Metrics  *ClusterMetrics
 }
 
+// NewVirtualClusterFromLocal return virtual cluster corresponding to local cluster
+func NewVirtualClusterFromLocal() *VirtualCluster {
+	return &VirtualCluster{
+		Name:     ClusterLocalName,
+		Type:     types.CredentialTypeInternal,
+		EndPoint: "-",
```
Comment (yue9944882) [inline pkg/multicluster/virtual_cluster.go:53]: nit: explicitly clarify the meaning of the "-" here, or define it as a named variable

File: test/e2e-multicluster-test/multicluster_test.go [hunk 3]
```diff
@@ -396,5 +398,33 @@ var _ = Describe("Test multicluster scenario", func() {
 			err = k8sClient.Delete(hubCtx, &checkApp)
 			Expect(err).Should(BeNil())
 		})
+
+		It("Test deploy multi-cluster application with target", func() {
+			By("apply application")
+			app := &v1beta1.Application{
+				ObjectMeta: v12.ObjectMeta{Namespace: namespace, Name: "test-app-target"},
```
Comment (yue9944882) [inline test/e2e-multicluster-test/multicluster_test.go:405]: nit: replace the generated import name to a elaborative named import such as "metav1", "clusterv1", etc

**Review by yue9944882 — COMMENTED** [review]
a few nits otherwise LGTM

**Review by barnettZQG — APPROVED** [review]
lgtm

**Review by FogDong — APPROVED** [review]

**Pull request merged**